beater/beater.go

```diff
@@ -46,9 +48,40 @@ type beater struct {
 	logger  *logp.Logger
 }
 
+var (
+	setupDashboardRemoved = errors.New("setting 'setup.dashboards' has been removed")
+)
+
+// checkConfig verifies the global configuration doesn't use unsupported settings
+func checkConfig(logger *logp.Logger) error {
+	cfg, err := cfgfile.Load("", nil)
+	if err != nil {
+		// responsibility for failing to load configuration lies elsewhere
+		// this is not reachable after going through normal beat creation
+		return nil
```

Comment:
if it is not reachable, maybe we could do without this `if` block. 
but even the case `err != nil` is covered already in L68 as `Unpack` would fail?

Reply:
Yeah, L68 would fail but I don't think we should rely on the behavior of `cfg` in case of an error.  This is reachable during tests so we definitely should keep the check.